### ISSUE TITLE
support flattening

### ISSUE THREAD

**Issue by tamird** [google/argh#48]
structopt supports "flattening", which allows one struct to be embedded in another "transparently". This is useful as documented in structopt, but also useful when two commands have the same arguments and the author does not wish to duplicate the code.

https://docs.rs/structopt/0.3.16/structopt/#flattening

**Comment by Boscop** [comment]
Yes please!
Without the ability to [chain subcommands](https://github.com/google/argh/issues/71), flattening is desperately needed.

Btw, this seems to be a duplicate of https://github.com/google/argh/issues/15

**Comment by tamird** [comment]
Agreed, looks like the same issue to me. This one has labels, so I'll leave it open until the maintainers do something with the other one.
